.aws/src/main.ts

```diff
@@ -114,18 +116,65 @@ class ListAPI extends TerraformStack {
     });
   }
 
+  /**
+   * Creates a serverless aurora RDS.
+   * This function should only be used when the environment is Dev
+   * @private
+   */
+  private createRds(vpc: PocketVPC) {
+    return new ApplicationRDSCluster(this, 'dev-aurora', {
+      prefix: `${config.prefix}-v1`,
+      vpcId: vpc.vpc.id,
+      subnetIds: vpc.privateSubnetIds,
+      rdsConfig: {
+        databaseName: config.name.toLowerCase(),
+        masterUsername: 'pkt_listapi',
+        skipFinalSnapshot: true,
+        engine: 'aurora-mysql',
+        engineMode: 'serverless',
+        scalingConfiguration: {
+          minCapacity: config.rds.minCapacity,
+          maxCapacity: config.rds.maxCapacity,
+          autoPause: false,
+        },
+        deletionProtection: false,
+      },
+      tags: config.tags,
+    });
+  }
+
   private createPocketAlbApplication(dependencies: {
     pagerDuty: PocketPagerDuty;
     region: datasources.DataAwsRegion;
     caller: datasources.DataAwsCallerIdentity;
     secretsManagerKmsAlias: kms.DataAwsKmsAlias;
     snsTopic: sns.DataAwsSnsTopic;
+    vpc: PocketVPC;
   }): PocketALBApplication {
-    const { pagerDuty, region, caller, secretsManagerKmsAlias, snsTopic } =
+    const { pagerDuty, region, caller, secretsManagerKmsAlias, snsTopic, vpc } =
       dependencies;
 
     const databaseSecretsArn = `arn:aws:secretsmanager:${region.name}:${caller.accountId}:secret:${config.name}/${config.environment}/READITLA_DB`;
 
+    /**
+     * Create an RDS instance if we are working in the Dev account.
+     * This is only to facilitate testing
+     */
+    let rdsCluster: ApplicationRDSCluster;
+
+    // Conditionally build secrets depending on environment
+    const secretResources = [
+      `arn:aws:secretsmanager:${vpc.region}:${vpc.accountId}:secret:${config.name}/${config.environment}`,
+      `arn:aws:secretsmanager:${vpc.region}:${vpc.accountId}:secret:${config.name}/${config.environment}/*`,
+    ];
```

Comment:
Do we need these here? Working off the example linked in the ticket.

Reply:
Yep! Since the RDS secret is not required for prod.

Reply:
Yep! Since the RDS secret is not required for prod.

Reply:
I think I misattributed your comment to the block below when it was actually intended for the block above.

the below part is already included in the taskExecutionRole secrets starting on line 303. You just need to conditionally add the dev RDS secret arn.

```
arn:aws:secretsmanager:${vpc.region}:${vpc.accountId}:secret:${config.name}/${config.environment}`,
      `arn:aws:secretsmanager:${vpc.region}:${vpc.accountId}:secret:${config.name}/${config.environment}/*`,
      ```